### PR TITLE
VOTE-346/updates english deadline dates (#575)

### PR DESCRIPTION
**Pull request by rayestrada**

### CODE REVIEW

**Review by mlloydbixal — APPROVED** [review]

File: data/translations/en/state-data.json [hunk 1]
```diff
@@ -269,13 +299,16 @@
     "more_info_link": "https://www.elections.il.gov/Default.aspx",
     "more_info_link_translated": "true",
     "more_info_link_language_selection": "",
-    "ip_deadline": "Available up to and including on Election Day",
-    "online_deadline": "16 days before Election Day",
-    "mail_received_deadline": "Must be received 28 days before Election Day",
-    "mail_postmarked_deadline": "",
+    "online_deadline": "Tuesday, October 18, 2022",
+    "ip_deadline": "Tuesday, November 8, 2022",
+    "mail_postmarked_deadline": "Must be postmarked by Thursday, October 06, 2022",
```
Comment (rayestrada) [inline data/translations/en/state-data.json:304]: Changed from postmarked to received. Which should we use? We can't use both.

File: data/translations/en/state-data.json [hunk 2]
```diff
@@ -341,13 +380,16 @@
     "more_info_link": "https://elect.ky.gov/Resources/Pages/Registration.aspx",
     "more_info_link_translated": "true",
     "more_info_link_language_selection": "",
-    "ip_deadline": "Fourth Monday before Election Day",
-    "online_deadline": "Fourth Monday before Election Day",
-    "mail_received_deadline": "Must be received by the fourth Monday before Election Day",
-    "mail_postmarked_deadline": "",
+    "online_deadline": "Tuesday, October 11, 2022",
+    "ip_deadline": "Tuesday, October 11, 2022",
+    "mail_postmarked_deadline": "Must be postmarked by Tuesday, October 11, 2022",
```
Comment (rayestrada) [inline data/translations/en/state-data.json:385]: Changed from received from postmarked. Which one should we use?

File: data/translations/en/state-data.json [hunk 3]
```diff
@@ -864,10 +964,11 @@
     "more_info_link": "https://www.vivote.gov/voters/register-vote",
     "more_info_link_translated": "true",
     "more_info_link_language_selection": "",
-    "ip_deadline": "30 days before Election Day",
+    "mail_postmarked_deadline": "Must be postmarked by Sunday, October 09, 2022",
```
Comment (rayestrada) [inline data/translations/en/state-data.json:967]: no `default_` value for postmarked deadline. Is this new?

File: data/translations/en/state-data.json [hunk 4]
```diff
@@ -749,27 +834,31 @@
     "more_info_link": "https://www.scvotes.gov/south-carolina-voter-registration-information",
     "more_info_link_translated": "true",
     "more_info_link_language_selection": "",
-    "ip_deadline": "30 days before Election Day",
-    "online_deadline": "30 days before Election Day",
-    "mail_received_deadline": "",
-    "mail_postmarked_deadline": "Must be postmarked 30 days before Election Day",
+    "online_deadline": "Sunday, October 09, 2022",
+    "ip_deadline": "Sunday, October 09, 2022",
+    "mail_received_deadline": "Must be received by Tuesday, October 11, 2022",
```
Comment (rayestrada) [inline data/translations/en/state-data.json:839]: Has both received and postmarked deadline. Which one should we use?

File: data/translations/en/state-data.json [hunk 5]
```diff
@@ -56,28 +61,31 @@
     "registration_link_language_selection": "",
     "more_info_link": "https://azsos.gov/elections/voting-election/register-vote-or-update-your-current-voter-information",
     "more_info_link_translated": "true",
-    "more_info_link_language_selection": "",
+    "more_info_link_language_selection": "Tuesday, October 11, 2022",
     "ip_deadline": "29 days before Election Day",
```
Comment (rayestrada) [inline data/translations/en/state-data.json:65]: `ip_deadline` not updated.

File: data/translations/en/state-data.json [hunk 6]
```diff
@@ -749,27 +834,31 @@
...
     "hp_link": "https://sdsos.gov/elections-voting/",
     "hp_link_translated": "true",
     "hp_link_language_selection": "",
     "more_info_link": "https://sdsos.gov/elections-voting/voting/register-to-vote/default.aspx",
     "more_info_link_translated": "true",
-    "more_info_link_language_selection": "",
-    "ip_deadline": "15 days before Election Day",
-    "mail_received_deadline": "Must be received 15 days before Election Day",
-    "mail_postmarked_deadline": "",
+    "more_info_link_language_selection": "Monday, October 24, 2022",
```
Comment (rayestrada) [inline data/translations/en/state-data.json:854]: This date was applied to the wrong property.

**Review by rayestrada — COMMENTED** [review]
@mlloydbixal  I have some requests for you to look at. Also, can you also remove the extra blank lines in the json objects.

File: data/translations/en/state-data.json [hunk 7]
```diff
@@ -233,13 +258,16 @@
     "more_info_link": "https://elections.hawaii.gov/voters/registration/",
     "more_info_link_translated": "true",
     "more_info_link_language_selection": "",
-    "ip_deadline": "Available up to and including on Election Day",
-    "online_deadline": "30 days before Election Day",
-    "mail_received_deadline": "Must be received 30 days before Election Day",
-    "mail_postmarked_deadline": "",
+    "ip_deadline": "Tuesday, November 8, 2022",
+    "online_deadline": "Tuesday, November 08, 2022",
+    "mail_postmarked_deadline": "Must be postmarked by Monday, October 31, 2022",
```
[the remark this answers is not in the document]
Comment (mlloydbixal) [reply data/translations/en/state-data.json:263]: Postmarked by is the updated version, so I will remove the fallback for received by. We still have it stored in state-data-2021.json

File: data/translations/en/state-data.json [hunk 8]
```diff
@@ -864,10 +964,11 @@
     "more_info_link": "https://www.vivote.gov/voters/register-vote",
     "more_info_link_translated": "true",
     "more_info_link_language_selection": "",
-    "ip_deadline": "30 days before Election Day",
+    "mail_postmarked_deadline": "Must be postmarked by Sunday, October 09, 2022",
     "confirm_registration_link": "https://www.vivote.gov/voters/lookup",
     "confirm_registration_link_translated": "true",
-    "confirm_registration_link_language_selection": ""
+    "confirm_registration_link_language_selection": "",
+    "default_ip_deadline": "30 days before Election Day"
```
[the remark this answers is not in the document]
Comment (mlloydbixal) [reply data/translations/en/state-data.json:971]: There is no new ip_deadline for this territory, I can follow up with Samira that it wasn't a mistake

File: data/translations/en/state-data.json (hunk 1 above)
Comment (mlloydbixal) [reply data/translations/en/state-data.json:304]: It changed to postmarked, we will not use the old received by

File: data/translations/en/state-data.json (hunk 3 above)
Comment (mlloydbixal) [reply data/translations/en/state-data.json:967]: Yes, it is new :)

File: data/translations/en/state-data.json (hunk 4 above)
Comment (mlloydbixal) [reply data/translations/en/state-data.json:839]: Updated to use only received

File: data/translations/en/state-data.json (hunk 5 above)
Comment (mlloydbixal) [reply data/translations/en/state-data.json:65]: updated

File: data/translations/en/state-data.json (hunk 6 above)
Comment (mlloydbixal) [reply data/translations/en/state-data.json:854]: updated

File: data/translations/en/state-data.json [hunk 9]
```diff
@@ -215,13 +238,15 @@
...
-    "mail_received_deadline": "Must be received 10 days before Election Day",
-    "mail_postmarked_deadline": "",
+    "ip_deadline": "Friday, October 28, 2022",
+    "mail_received_deadline": "Must be received by Friday, October 28, 2022",
     "confirm_registration_link": "https://gec.guam.gov/validate/",
     "confirm_registration_link_translated": "true",
-    "confirm_registration_link_language_selection": ""
+    "confirm_registration_link_language_selection": "",
+    "default_ip_deadline": "10 days before Election Day",
+    "default_online_deadline": "10 days before Election Day",
```
[the remark this answers is not in the document]
Comment (mlloydbixal) [reply data/translations/en/state-data.json:247]: We do not have an updated online deadline for Guam. If we know we aren't waiting for one I will remove the fallback so no online deadline is used

[the remark this answers is not in the document]
Comment (SamiraMSadat) [reply data/translations/en/state-data.json]: As of today, there is no online registration date for Guam. They might have had one in the past, but their online portal is not open or functioning. The territories are a bit tricky and might change last minute. I am tracking that and I will take charge of any future changes.

File: data/translations/en/state-data.json (hunk 7 above)
[the remark this answers is not in the document]
Comment (SamiraMSadat) [reply data/translations/en/state-data.json:262]: Confirmed

File: data/translations/en/state-data.json (hunk 8 above)
[the remark this answers is not in the document]
Comment (SamiraMSadat) [reply data/translations/en/state-data.json:963]: @mlloydbixal Good catch! This is my mistake. I forgot to update from the EAC spreadsheet to our spreadsheet for two territories. I highlighted the U.S. Virgin Islands and Northern Marina Islands rows. Here is a source for extra confirmation: https://www.vivote.gov/voters/register-vote

File: data/translations/en/state-data.json (hunk 1 above)
Comment (SamiraMSadat) [reply data/translations/en/state-data.json:301]: confirmed

File: data/translations/en/state-data.json (hunk 2 above)
Comment (SamiraMSadat) [reply data/translations/en/state-data.json:380]: confirmed

File: data/translations/en/state-data.json (hunk 3 above)
Comment (SamiraMSadat) [reply data/translations/en/state-data.json:958]: confirmed

File: data/translations/en/state-data.json (hunk 4 above)
Comment (SamiraMSadat) [reply data/translations/en/state-data.json:832]: confirmed

File: data/translations/en/state-data.json (hunk 9 above)
[the remark this answers is not in the document]
Comment (rayestrada) [reply data/translations/en/state-data.json]: Okay we will keep the fallback for now. It appears like on Guam's website https://gec.guam.gov/ registration is closed until August 28th

File: data/translations/en/state-data.json [hunk 10]
```diff
@@ -244,9 +244,8 @@
     "confirm_registration_link_translated": "true",
     "confirm_registration_link_language_selection": "",
     "default_ip_deadline": "10 days before Election Day",
-    "default_online_deadline": "10 days before Election Day",
```
Comment (rayestrada) [inline data/translations/en/state-data.json]: We should keep this one since registration will be open in a few weeks online.

**Review by rayestrada — COMMENTED** [review]
I left one comment for Guam but everything else looks good.

**Comment by SamiraMSadat** [comment]
Massachusetts dates are incorrect. They changed from 20 days prior to election day to 10 days (Saturday, October 29, 2022). Please view on the spreadsheet and change in the code.
https://docs.google.com/spreadsheets/d/15sasvySV1wC3558GJwHxrVUsEmyN1YlAVmzunErH9jw/edit#gid=1917147583

**Comment by SamiraMSadat** [comment]
I just noticed Northern Mariana Islands and Puerto Rico are missing the mail in deadline. Also U.S. Virgin Islands "In person registration" is October 9.  I think we updated this in the spreadsheet while you were out of office and I forgot to follow up. My apologies! The rows in question are highlighted.

**Comment by rayestrada** [comment]
@SamiraMSadat can you check this again and approve the changes if they are complete?

**Review by SamiraMSadat — DISMISSED** [review]

**Pull request merged**